nncf/experimental/onnx/model_normalizer.py

```diff
@@ -80,44 +87,64 @@ def add_const_value_infos_to_graph(graph: onnx.GraphProto):
                         for g in attr.graphs:
                             add_const_value_infos_to_graph(g)
 
-        return add_const_value_infos_to_graph(model.graph)
+        modified_model = deepcopy(model)
+        add_const_value_infos_to_graph(modified_model.graph)
+        onnx.checker.check_model(modified_model)
+        return modified_model
+
+    @staticmethod
+    def infer_models_shape(model: onnx.ModelProto) -> onnx.ModelProto:
+        """
+        Infers 'model' and saves the edges shape into the 'model'.
+        """
+        # ONNX shape inference
+        # https://github.com/onnx/onnx/blob/main/docs/proposals/SymbolicShapeInfProposal.md
+        onnx.checker.check_model(model)
+        inferred_shape_model = onnx.shape_inference.infer_shapes(model)
+        onnx.checker.check_model(inferred_shape_model)
+        return inferred_shape_model
 
     @staticmethod
-    def convert_opset_version(model: onnx.ModelProto) -> onnx.ModelProto:
+    def convert_opset_version(model: onnx.ModelProto, opset_version: int,
+                              ir_version: int) -> onnx.ModelProto:
         """
-        Try to convert model to opset version 13, also adding some important information to the model such shapes.
+        Converts 'model' Opset Version to 'opset_version' if possible and changes IR Version to 'ir_version'.
+        If the 'model' can not be converted returns the original 'model'.
         """
         # pylint: disable=no-member
         onnx.checker.check_model(model)
-        nncf_logger.info('Original opset = {}'.format(model.opset_import[0].version))
-        nncf_logger.info('Original ir_version = {}'.format(model.ir_version))
+        model_opset = model.opset_import[0].version
+        model_ir_version = model.ir_version
+        nncf_logger.debug('Original Opset Version = {}'.format(model_opset))
+        nncf_logger.debug('Original IR Version = {}'.format(model_ir_version))
 
         try:
             modified_model = deepcopy(model)
-            modified_model.ir_version = 7  # Due to the 'Shufflenet-v1
-            modified_model = convert_version(modified_model, 13)
-
-            # ONNX shape inference
-            # https://github.com/onnx/onnx/blob/main/docs/proposals/SymbolicShapeInfProposal.md
-            modified_model = onnx.shape_inference.infer_shapes(modified_model)
-            ONNXModelNormalizer.add_input_from_initializer(modified_model)
-
+            if model_ir_version < ir_version:
+                # TODO(kshpv): is it legal to change ir_vesrsion?
```

Comment:
You should find this out before merging this PR, and to be sure about this, and leave a comment on whether it is legal (or at least doesn't break anything) or not to change `ir_version` and in what circumstances.

Reply:
Done